fjord/heartbeat/healthchecks.py

```diff
@@ -0,0 +1,205 @@
+from collections import Counter, namedtuple
+from datetime import datetime, timedelta
+import logging
+
+from django.conf import settings
+from django.core.mail import send_mail
+from django.db import connection
+
+from fjord.base.email_utils import render_email
+from fjord.heartbeat.models import Answer
+from fjord.mailinglist.utils import get_recipients
+
+
+log = logging.getLogger('i.heartbeat')
+
+
+SEVERITY_LOW = 1
+SEVERITY_MEDIUM = 5
+SEVERITY_HIGH = 10
+
+SEVERITY = {
+    SEVERITY_LOW: 'low',
+    SEVERITY_MEDIUM: 'medium',
+    SEVERITY_HIGH: 'high'
+}
+
+
+Result = namedtuple('Result', ['name', 'severity', 'summary', 'output'])
+
+
+CHECKS = []
+
+
+def register_check(cls):
+    CHECKS.append(cls)
+    return cls
+
+
+class Check(object):
+    name = ''
+
+    @classmethod
+    def check(cls):
+        pass
+
+
+@register_check
+class CheckAnyAnswers(Check):
+    """Are there any heartbeat answers? If not, that's very bad."""
+    name = 'any answers check'
+
+    @classmethod
+    def check(cls):
+        day_ago = datetime.now() - timedelta(days=1)
+        count = Answer.objects.filter(received_ts__gt=day_ago).count()
+
+        if count == 0:
+            return Result(
+                cls.name,
+                SEVERITY_HIGH,
+                '0 answers in last 24 hours.',
+                str(count)
+            )
+        return Result(
+            cls.name,
+            SEVERITY_LOW,
+            '%s answers in last 24 hours.' % str(count),
+            str(count)
+        )
+
+
+def tableify(table):
+    num_cols = 0
+    maxes = []
+
+    for row in table:
+        num_cols = max(num_cols, len(row))
+        if len(maxes) < len(row):
+            maxes.extend([0] * (len(row) - len(maxes)))
+        for i, cell in enumerate(row):
+            maxes[i] = max(maxes[i], len(str(cell)))
+
+    def fix_row(maxes, row):
+        return '  '.join([
+            str(cell) + (' ' * (maxes[i] - len(str(cell))))
+            for i, cell in enumerate(row)
+        ])
+
+    return '\n'.join(
+        [
+            fix_row(maxes, row)
+            for row in table
+        ]
+    )
+
+
+@register_check
+class CheckMissingVotes(Check):
+    """should freak out if votes are 0 for any 'large' cell."""
```

Comment:
I have no idea. That's part of the reason I have no idea what a high severity issue would look like here. I need to talk to Gregg, but he's out this week.